src/python/tritonfrontend/CMakeLists.txt

```diff
@@ -96,13 +94,10 @@ endif()
 
 if(${TRITON_ENABLE_TRACING})
   message("TRACING/STATS IS CURRENTLY NOT SUPPORTED.")
```

Comment:
I don't understand this part, if not supported, then why introducing the dependency?

Reply:
Currently, when building through `build.py` or `cmake`, tracing can be enabled in the build. From my understanding, this means that `tracer.h` will get built with the `#ifdef TRITON_ENABLE_TRACING` [sections included](https://github.com/triton-inference-server/server/blob/fde6e5887775b1236e25192601b6cd1d7abe3620/src/tracer.h#L40C1-L57C1). Hence, when `http-endpoint-library` or `grpc-endpoint-library` include `tracer.h` ([here](https://github.com/triton-inference-server/server/blob/fde6e5887775b1236e25192601b6cd1d7abe3620/src/http_server.h#L44C1) and [here](https://github.com/triton-inference-server/server/blob/fde6e5887775b1236e25192601b6cd1d7abe3620/src/grpc/grpc_server.h#L35)) it will look for the tracing dependencies that `tracer.h` expects

Reply:
> Hence, when http-endpoint-library or grpc-endpoint-library include tracer.h ([here](https://github.com/triton-inference-server/server/blob/fde6e5887775b1236e25192601b6cd1d7abe3620/src/http_server.h#L44C1) and [here](https://github.com/triton-inference-server/server/blob/fde6e5887775b1236e25192601b6cd1d7abe3620/src/grpc/grpc_server.h#L35)) it will look for the tracing dependencies that tracer.h requires.

Then shouldn't `tracing-library` be added as `http-endpoint-library` / `grpc-endpoint-library`'s dependencies?

Reply:
I think this would be a good change. Currently instead of linking the `tracing-library`, we do something along the lines of:
http_server.h
```cpp
...
#include "tracer.h"
...
```
and in CMakeList.txt to handle the dependencies we do:
```cmake
target_include_directories(
      http-endpoint-library
      PRIVATE ${OPENTELEMETRY_CPP_INCLUDE_DIRS}
    )
```
But now that we are changing the `tracing-library` links/includes to be public, I replaced:
```cmake
target_include_directories(
main/http-endpoint-library/grpc-endpoint-library
PRIVATE 
${OPENTELEMETRY_CPP_INCLUDE_DIRS})
```
with this:
```cmake
target_link_libraries(
main/http-endpoint-library/grpc-endpoint-library
PRIVATE
tracing-library)
```

Reply:
With assistance from @fpetrini15, I was able to verify that these changes work with the windows build as well.
CI Pipeline ID: 19374177